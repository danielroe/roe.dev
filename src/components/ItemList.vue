<template functional>
  <section :class="[$style.list, data.staticClass, data.class]">
    <slot />
  </section>
</template>

<style lang="postcss" module>
.list {
  @apply flex flex-row flex-wrap -mx-2;
  > * {
    @apply p-4 m-2 relative;
    @apply text-xl flex flex-col justify-end;
    @apply shadow;

    background-color: var(--accent, theme('colors.gray.900'));
    min-height: 10rem;
    flex: 40%;
    @media (width < 767px) {
      flex: 100%;
      min-height: 3rem;
    }

    transition: 0.3s transform, 0.3s border-color;
    border: 1px solid transparent;

    &:hover,
    &:active,
    &:focus {
      @apply outline-none;

      border-color: var(--text-muted);
    }
    &::after {
      @apply text-transparent;
    }
  }
}
</style>
