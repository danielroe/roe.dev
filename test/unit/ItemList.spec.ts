import { shallowMount } from '@vue/test-utils'
import ItemList from '~/components/ItemList.vue'

describe('ItemList.vue', () => {
  it('renders slot', () => {
    const msg = 'new message'
    const wrapper = shallowMount(ItemList, {
      slots: {
        default: msg,
      },
    })
    expect(wrapper.text()).toMatch(msg)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
