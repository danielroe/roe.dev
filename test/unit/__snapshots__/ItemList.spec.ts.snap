// Vitest Snapshot v1

exports[`ItemList.vue > renders slot 1`] = `"<section class=\\"_list_e30301\\">new message</section>"`;
